Things You Should Know

btardif opened the pull request:
FYI @stefsch 

I will need a code review for this before I merge it in. Will work with Linux team to get SME coverage

EricSten-MSFT commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+
+**App Service** will wait 230 seconds for a container to start, initialize and respond to an HTTP request. **App Service** interprets this first response to an HTTP request as an idication that your app has initialized and is ready to recive further requests. If the timout of 230 seconds is reached with no response from the app, then it's considerd a failure to start. 
+
+Some containers might need more than 230 seconds to start, the default timeout can be overridden through the `WEBSITES_CONTAINER_START_TIME_LIMIT` app setting. This app setting can be set to a maximum of 1800 seconds.
```
It would be nice to mention that customers should really investigate why apps are taking longer than 230 seconds to start....that could be the sign of something worse.  Customers should review memory and CPU consumption, and avoid doing large DB queries at initialization.

EricSten-MSFT commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+
+You're probably used to using `df` to find out disk space in your volumes. You can certainly do that in **App Service**, but what you'll see isn't really going to be as helpful as you might think.
+
+Assuming you are persisting files in the /home directory (which is currently the default), you will have a quota based on your pricing tier. However, if you use `df`, the disk space you see **won't reflect that quota**. If you want to find out your disk usage, use **Quotas** in your app's menu, **File System Storage** in the **App Service Plan** menu, or use the REST API to find your usage.
```
This is a super important point!  Customers using df will think there's a petabyte of space, when in reality they only have 10GB or less.  Further, there's no way to find out what the quota is, or how close they are to the limit.

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+
+## General Information
+
+### You can requests features to the App Service team
```
Misspelling  - should be "You can request features...."

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+
+## Deployment
+
+### Web Deploy
```
I would remove this section - Web Deploy to Linux App Service isn't really a deployment model that we want to be pushing.

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+|:------:|:-----------:|
+| ⛔    | ✅          |
+
+The first time you use a custom Docker image, we will do a `docker pull` and pull all layers. These layers are stored on disk just as if you were using Docker on-premise. When we do a `docker pull`  after a app restart, we will only pull layers that have changed. If there have been no changes, we will simply use existing layers on disk.
```
"after an app restart"

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+
+If you don't require file persistence, you can set this app setting to **false**. this will make your app resilient to storage failovers.
+
+The absence of this app setting will result in the setting being **"true"**. In other words, if this app setting does not exist for your app storage will be mounted.
```
This feels like a roundabout way of saying that by default when you create a containerized app on App Service we automatically mount /home to App Service's remote storage.

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+
+The absence of this app setting will result in the setting being **"true"**. In other words, if this app setting does not exist for your app storage will be mounted.
+
+[**Kudu** (Advanced Tools)](https://github.com/projectkudu/kudu/wiki) for Linux Apps runs in a separate container. The **Kudu** always maps the /home directory to the **App Service** built in storage.  That way, the `/home/LogFiles` directory will persist between restarts and scale out operations.
```
This section is out of place relative to a topic about storage persistence.

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+|:------:|:-----------:|
+| ✅    | ✅          |
+
+By default, we don't persist files in your app. However, if you need to persist files, you can enable **App Service** built in storage (see [You can enable and disable storage persistence with an app setting](You-can-enable-and-disable-storage-persistence-with-an-app-setting)). 
```
This contradicts the earlier section about enabling persistent storage.  Also for bring your own code we certainly do persist files written to /home.

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+
+If you enable this, any files saved within the /home directory and sub-directories will persist. However, any files or directories outside of the /home directory will not be persisted between app recycles. This can impact your app in a couple of different ways:
+
+- If you use npm, apt, or similar to install something from the console after your app starts, once the app restarts, whatever you installed will be gone.
```
I would change this comment since with Oryx we install quite a bit on behalf of the app and it does persist across restarts.  I think the current explanation/recommendation would be to use the built in build system for bring your own code rather than attempting to manually run npm, apt, etc... from a console for precisely the reason that the materialized packages will all be lost after an application restart.

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+|:------:|:-----------:|
+| ⛔    | ✅          |
+
+**App Service** will a certain amount of time for your container to start before we consider it be a failed start. In order for **App Service** to consider a container to be successfully started, the container must start and it must respond to an HTTP ping. If the container starts but does not respond to a ping, we will eventually log an event in the Docker log saying that it didn't start.
```
Http ping health check on Linux:
a.	By default, we ping the / of the app and ensure its returning some valid http status code at the start of the container – if the container doesn’t respond within 4 minutes, we stop the container. Note that this specific check is only done during the cold start.   Note - I don't know if this cold start health ping is configurable on our platform.
b.	URL specified by WEBSITE_HEALTH_CHECK_URL – This is configurable by the customer - We will ping this URL and if the container doesn’t respond 200 for ~20 seconds, we will stop the container and restart the site. This specific check is done on a timer thread.

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+|:------:|:-----------:|
+| ✅    | ✅          |
+
+**App Service** terminates SSL/TLS at the front-ends. That means that SSL/TLS requests never get to your app. That's good news for you because it means that you don't need to (and should not) implement any support for SSL/TLS into your app. Also as stated above, it's important to understand that the front-ends where SSL/TLS is terminated are inside of our Azure data centers. If you use SSL/TLS wi ...
```
Confusing wording:  "If you use SSL/TLS in your app to make outbound calls,...."

stefsch commented on Things_You_Should_Know/things_you_should_know.md:
```diff
@@ -0,0 +1,400 @@
+
+We offer a **BASH** console in [Kudu (Advanced Tools)](https://github.com/projectkudu/kudu/wiki) that you can use to browse directory structure, etc. It's important to realize that this console is running inside of **Kudu**. This is not the same container that your app is running in, so if you look at files and directories here, they won't reflect your app. The only exception to that is the /home ...
+
+### You don't need (and should not have) SSL/TLS support in your app
```
The title is misleading.  Should be more like "App Service TLS/SSL Support for your app"

btardif commented:
Thanks to @stefsch and @EricSten-MSFT for review comments, I think I have addressed all the feedback. I'm going to wait till Friday to merge this in case anyone else form Linux team can take a pass.
